example_configs/sssd.md

```diff
@@ -0,0 +1,149 @@
+# SSSD configuration
+
+> [!WARNING]  
+> Since we need to create custom user attributes, you must run `latest` version of LLDAP - `stable` does not have this functionality yet.
+
+The following configuration was tested on sssd version 2.9.1. It may not work on older versions.
+
+To work properly SSSD also needs corret openldap-client configuration.
+
+## Packages
+
+Here are the packages needed in Oracle Linux 9:
+
+`openldap-clients openldap-devel sssd sssd-ldap`
+
+
+## Configure OpenLDAP client
+
+### Certificates - skip if you don't use LDAPS
+
+1. Place your certs in `/etc/openldap/certs`
+2. Run `openssl rehash /etc/openldap/certs`
+
+### Config
+
+Create or modify `/etc/openldap/ldap.conf`:
+```
+URI ldaps://LDAP_SERVER_URL # Use ldap:// if you don't use LDAPS
+BASE dc=example,dc=com
+TLS_CACERTDIR /etc/openldap/certs # Skip if you don't use LDAPS
+ssl on # Skip if you don't use LDAPS
+```
+
+### Verify
+
+To check if your config is correct you can use `ldapsearch`:
+
+```bash
+ldapsearch -x -D "uid=admin,ou=people,dc=example,dc=com" -w "${ADMIN_PASSWORD}" -b "ou=people,dc=example,dc=com"
+```
+
+The above command should return list of your users.
+
+## Prepare LLDAP
+
+SSSD requires some specific setup.
+
+All of your users need `uidnumber` and `gidnumber` - these are not generated by default in LLDAP.
+
+To create these user attributes you can use either the web GUI or [lldap-cli](https://github.com/Zepmann/lldap-cli), but webp GUI does not support setting them.
+
+To create these custom user attributes with [lldap-cli](https://github.com/Zepmann/lldap-cli):
+
+```bash
+# Login to your LLDAP server
+eval $(lldap-cli -D admin -w abcd1234 login)
+
+# Add uidnumber and gidnumber attributes
+lldap-cli schema attribute user add uidnumber integer
+lldap-cli schema attribute user add gidnumber integer
+```
+
+You must also manually set the above attributes for each user, eg. for user `admin` with `uidnumber` 2000 and `gidnumber` 2000:
+```bash
+lldap-cli user update set admin uidnumber 2000
+lldap-cli user update set admin gidnumber 2000
```

Comment:
Well now I’m wondering, but I’m pretty sure user needs gidNumber as well. Going to check that once I’m able to setup a test environment.

Reply:
Yeah, I just checked, every user needs a gid for their primary group. A common strategy is to have a group per user so they don't accidentally share files by making them group-readable. I'm not sure it's a good idea here, though (you probably don't want to create an LDAP group per user).

It doesn't matter (yet) since you assign the gid by hand, but it's something to think about. Maybe you should have a note about this? Something about setting the default file mask to 700 if you don't want every user to be able to read your files.

Reply:
Well, what I did was one common group (gid 2000) which is also a primary group for every user. This was desired in my case, at my LDAP instance is for a team of people performing common tasks and are sharing the files (that's why I also hardcoded the default home directory). I also handle creation and setting permissions for the home directory separately (in a configuration script).

Reply:
Yeah, I still think it's worth pointing out that this is a design decision that the user needs to make, and to explain the two options (one group per user, or one shared group), as well as the default file mask.